orm/hibernate-orm-4/src/test/java/org/hibernate/bugs/ORMStandaloneJpaUnitTestCase.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2014 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hibernate.bugs;
+
+import org.hibernate.cfg.Environment;
+import org.hibernate.internal.util.StringHelper;
+import org.hibernate.jpa.AvailableSettings;
+import org.hibernate.jpa.HibernatePersistenceProvider;
+import org.hibernate.jpa.boot.spi.Bootstrap;
+import org.hibernate.jpa.boot.spi.PersistenceUnitDescriptor;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.SharedCacheMode;
+import javax.persistence.ValidationMode;
+import javax.persistence.spi.PersistenceUnitTransactionType;
+import java.net.URL;
+import java.util.*;
+
+/**
+ * This template demonstrates how to develop a standalone test case for Hibernate ORM using the JPA specification.
+ */
+public class ORMStandaloneJpaUnitTestCase {
+
+	private EntityManagerFactory entityManagerFactory;
+
+    // Add your entities here.
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[0];
+	}
+
+    // If you use *.hbm.xml mappings, instead of annotations, add the mappings here.
+    protected String[] getMappings() {
+        return new String[0];
+    }
+
+	// Add your tests, using standard JUnit.
+	@Test
+	public void hhh123Test() throws Exception {
+
+	}
+
+    @Before
+    public void setup() {
+        entityManagerFactory =  Bootstrap.getEntityManagerFactoryBuilder(
+                buildPersistenceUnitDescriptor(),
+                buildSettings()
+        ).build();
+    }
+
+	private PersistenceUnitDescriptor buildPersistenceUnitDescriptor() {
```

Comment:
That's the approach taken from hibernate-entitymanager tests and [I've been using the same approach too](http://vladmihalcea.com/2015/11/26/how-to-bootstrap-hibernate-without-the-persistence-xml-file/). The advantage is when you need tons of such tests and you don't want to share anything from one test to the other. You just want each test to be completely isolated and self-contained. It's same thing with the native tests as well. Instead of configuring everything in a `hibernate.cfg.cml`, it's more flexible to supply everything in the tets case itself.